### PR TITLE
errorhist and sun

### PR DESCRIPTION
**Pull request by dneise**
* add errorhist; 
 * adjust sun(), so it does the same as smartfact.cc intends.

Should be "tested" tonight.

### CODE REVIEW

File: smart_fact_crawler/__init__.py [hunk 1]
```diff
@@ -124,24 +125,36 @@ def sqm(url=smartfacturl + 'sqm.data'):
 
 def sun(url=smartfacturl + 'sun.data'):
 
-    def hhmm2ms(hhmm):
-        return int(hhmm[0:2]) * 3600 * 1000 + int(hhmm[3:5]) * 60 * 1000
+    def next_datetime_from_hhmm_string(hhmm):
+        now = datetime.utcnow()
+        now_tuple = now.timetuple()
```
Comment (MaxNoe) [inline smart_fact_crawler/__init__.py]: This is unused

File: smart_fact_crawler/__init__.py [hunk 2]
```diff
@@ -124,24 +125,36 @@ def sqm(url=smartfacturl + 'sqm.data'):
...
-        'Start_of_dark_time':       smartfact_time2datetime(date_ms + hhmm2ms(tc[8, 1])),
+        '0Time_Stamp': smartfact_time2datetime(tc[0, 0]),
+        '1End_of_dark_time':         next_datetime_from_hhmm_string(tc[1, 1]),
+        '2End_of_astro_twilight':    next_datetime_from_hhmm_string(tc[2, 1]),
+        '3End_of_nautic_twilight':   next_datetime_from_hhmm_string(tc[3, 1]),
+        '4Start_of_day_time':        next_datetime_from_hhmm_string(tc[4, 1]),
+        '5End_of_day_time':          next_datetime_from_hhmm_string(tc[5, 1]),
+        '6Start_of_nautic_twilight': next_datetime_from_hhmm_string(tc[6, 1]),
+        '7Start_of_astro_twilight':  next_datetime_from_hhmm_string(tc[7, 1]),
+        '8Start_of_dark_time':       next_datetime_from_hhmm_string(tc[8, 1]),
```
Comment (MaxNoe) [inline smart_fact_crawler/__init__.py]: Do we really want these numbers in front?

File: smart_fact_crawler/__init__.py (hunk 1 above)
Comment (dneise) [inline smart_fact_crawler/__init__.py]: yes thanks

File: smart_fact_crawler/__init__.py (hunk 2 above)
Comment (dneise) [inline smart_fact_crawler/__init__.py]: nope this is shit I needed to sort the string repr of the dict? so it looks the way I like it.

was an error to leave it in

**Comment by dneise** [comment]
can this be merged?

**Comment by MaxNoe** [comment]
Any objections to merging?

**Comment by dneise** [comment]
Does this conflict with #5 ? 
Should we first merge this and then merge #5?

**Comment by MaxNoe** [comment]
Maybe first 5 and then check if this still works?

**Comment by dneise** [comment]
yes

**Comment by dneise** [comment]
Okay, so I tested it and it works all for me ... i.e. it doesn't throw up

**Pull request merged**